app/src/main/java/org/jellyfin/androidtv/details/presenters/SeriesMediaRowPresenter.kt

```diff
@@ -0,0 +1,166 @@
+package org.jellyfin.androidtv.details.presenters
+
+import android.content.Context
+import android.text.format.DateFormat
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import android.widget.Toast
+import androidx.leanback.widget.RowPresenter
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.row_details_series_media.view.*
+import kotlinx.android.synthetic.main.row_details_series_media_item.view.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.TvApp
+import org.jellyfin.androidtv.details.rows.SeriesMediaRow
+import org.jellyfin.androidtv.model.itemtypes.Episode
+import org.jellyfin.androidtv.model.itemtypes.Season
+import org.jellyfin.androidtv.ui.RecyclerViewSpacingDecoration
+import org.jellyfin.androidtv.util.TimeUtils
+import org.jellyfin.androidtv.util.apiclient.getEpisodesOfSeason
+import org.jellyfin.androidtv.util.dp
+
+class SeriesMediaRowPresenter(
+	private val context: Context
+) : RowPresenter() {
+	init {
+		headerPresenter = null
+		selectEffectEnabled = false
+	}
+
+	class ViewHolder(view: View) : RowPresenter.ViewHolder(view) {
+		val seasonList = view.season_list.apply {
+			adapter = SeasonListAdapter()
+			addItemDecoration(RecyclerViewSpacingDecoration(16.dp))
+		}
+		val poster = view.poster
+		val seasonContainer = view.season_container.apply {
+			adapter = SeasonContainerAdapter()
+			addItemDecoration(RecyclerViewSpacingDecoration(16.dp))
+		}
+	}
+
+	override fun createRowViewHolder(parent: ViewGroup): ViewHolder {
+		val view = LayoutInflater
+			.from(parent.context)
+			.inflate(R.layout.row_details_series_media, parent, false)
+
+		return ViewHolder(view)
+	}
+
+	override fun onBindRowViewHolder(viewHolder: RowPresenter.ViewHolder, row: Any) {
+		viewHolder as ViewHolder
+		row as SeriesMediaRow
+		val series = row.item
+
+		(viewHolder.seasonList.adapter as SeasonListAdapter).apply {
+			setItems(row.seasons)
+			setItemChangeListener {
+				setSelectedSeason(viewHolder, it)
+			}
+		}
+	}
+
+	private fun setSelectedSeason(vh: ViewHolder, item: Season) {
+		item.images.primary?.load(context) {
+			vh.poster.setImageBitmap(it)
+		}
+
+		Toast.makeText(context, item.title, Toast.LENGTH_LONG).show()
+		GlobalScope.launch(Dispatchers.IO) {
```

Comment:
`GlobalScope` is not recommended (please see this post: https://medium.com/@elizarov/the-reason-to-avoid-globalscope-835337445abc). It's better to use either a "built-in" scope (in Activity, Fragment or ViewModel) or make your own by extending CoroutineScope interface.